File: Doubly Linked List/BasicDLL.py
class Node: 
    def __init__(self,data):
        self.data = data
        self.next = None
        self.prev = None

node1 = Node(5)
node2 = Node(7)
node3 = Node(11)
node4 = Node(14)

node1.next = node2

node2.prev = node1
node2.next = node3

node3.next= node4
node3.prev = node2

node4.prev= node3

print("\nTraversing forward:")
currentNode = node1
while currentNode:
    print(currentNode.data, end=" -> ")
    currentNode = currentNode.next
print('null')

print("\nTraversing backward:")
currentNode = node4
while currentNode:
    print(currentNode.data , end=' -> ')
    currentNode = currentNode.prev
print("null")
        

File: Doubly Linked List/BasicDLL.c
#include<stdio.h>
#include<stdlib.h>


typedef struct Node {
    int data;
    struct Node *next;
    struct Node *prev;
}Node;

int main(){
    Node *node1 = (Node*)malloc(sizeof(Node));
    node1->data = 3;
    node1->next = NULL;
    node1->prev = NULL;

    Node *node2 = (Node*)malloc(sizeof(Node));
    node2->data = 5;
    node2->next = NULL;
    node2->prev = node1;
    node1->next = node2;

    Node *node3 = (Node*)malloc(sizeof(Node));
    node3->data = 13;
    node3->next = NULL;
    node3->prev = node2;
    node2->next = node3;

    Node *node4 = (Node*)malloc(sizeof(Node));
    node4->data = 2;
    node4->next = NULL;
    node4->prev = node3;
    node3->next = node4;

    Node *currentNode = node1;
    printf("Forward: ");
    while (currentNode)
    {
        printf("%d ->", currentNode->data);
        currentNode = currentNode->next;
    }
    printf("NULL\n");

    currentNode = node4;
    printf("Backward: ");
    while (currentNode)
    {
        printf("%d ->", currentNode->data);
        currentNode = currentNode->prev;
    }
    printf("NULL\n");

    free(node1);
    free(node2);
    free(node3);
    free(node4);

    return 0;
    
}